devito/types/dimension.py

```diff
@@ -753,9 +753,15 @@ class ConditionalDimension(DerivedDimension):
     is_NonlinearDerived = True
     is_Conditional = True
 
-    def __init_finalize__(self, name, parent, factor=None, condition=None,
+    def __init_finalize__(self, name, parent=None, factor=None, condition=None,
                           indirect=False):
+        # `parent=None` degenerates to a ConditionalDimension outside of
+        # any iteration space
```

Comment:
outside of any iteration space -->
with an empty IterationSpace
?

Reply:
no, we obv don't know about IterationSpaces here